openedx/core/djangoapps/notifications/tasks.py

```diff
@@ -194,34 +203,37 @@ def send_notifications(user_ids, course_key: str, app_name, notification_type, c
                     course_id=course_key,
                     web='web' in notification_preferences,
                     email=email_enabled,
+                    push=push_notification,
                     group_by_id=group_by_id,
                 )
                 if email_enabled and (email_cadence == EmailCadence.IMMEDIATELY):
                     email_notification_mapping[user_id] = new_notification
 
+                if push_notification:
+                    push_notification_audience.append(user_id)
+
                 if grouping_enabled and existing_notifications.get(user_id, None):
                     group_user_notifications(new_notification, existing_notifications[user_id])
-                    if not notifications_generated:
-                        notifications_generated = True
-                        notification_content = new_notification.content
+                    if not generated_notification:
+                        generated_notification = new_notification
                 else:
                     notifications.append(new_notification)
                 generated_notification_audience.append(user_id)
 
         # send notification to users but use bulk_create
         notification_objects = Notification.objects.bulk_create(notifications)
-        if notification_objects and not notifications_generated:
-            notifications_generated = True
-            notification_content = notification_objects[0].content
+        if notification_objects and not generated_notification:
+            generated_notification = notification_objects[0]
 
     if email_notification_mapping:
         send_immediate_cadence_email(email_notification_mapping, course_key)
 
-    if notifications_generated:
+    if generated_notification:
         notification_generated_event(
             generated_notification_audience, app_name, notification_type, course_key, content_url,
-            notification_content, sender_id=sender_id
+            generated_notification.content, sender_id=sender_id
         )
+        send_ace_msg_to_push_channel(push_notification_audience, generated_notification, sender_id)
```

Comment:
What other types of notifications are generated by this code? Email and the in-platform notification tray, I assume?

Why do you need a special function call (`send_ace_msg_to_push_channel`) in order to send the push notification, but not to send the other kinds of notifications?